app/code/Magento/MediaGallery/Model/ResourceModel/Keyword/GetAssetsKeywords.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallery\Model\ResourceModel\Keyword;
+
+use Magento\Framework\Exception\IntegrationException;
+use Magento\MediaGalleryApi\Api\Data\AssetKeywordsInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetKeywordsInterfaceFactory;
+use Magento\MediaGalleryApi\Api\Data\KeywordInterface;
+use Magento\MediaGalleryApi\Api\Data\KeywordInterfaceFactory;
+use Magento\MediaGalleryApi\Api\GetAssetsKeywordsInterface;
+use Magento\Framework\App\ResourceConnection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Retrieve keywords of the media assets
+ */
+class GetAssetsKeywords implements GetAssetsKeywordsInterface
+{
+    private const TABLE_KEYWORD = 'media_gallery_keyword';
+    private const TABLE_ASSET_KEYWORD = 'media_gallery_asset_keyword';
+    private const FIELD_ASSET_ID = 'asset_id';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var KeywordInterfaceFactory
+     */
+    private $keywordFactory;
+
+    /**
+     * @var AssetKeywordsInterfaceFactory
+     */
+    private $assetKeywordsFactory;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param AssetKeywordsInterfaceFactory $assetKeywordsFactory
+     * @param ResourceConnection $resourceConnection
+     * @param KeywordInterfaceFactory $keywordFactory
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        AssetKeywordsInterfaceFactory $assetKeywordsFactory,
+        ResourceConnection $resourceConnection,
+        KeywordInterfaceFactory $keywordFactory,
+        LoggerInterface $logger
+    ) {
+        $this->assetKeywordsFactory = $assetKeywordsFactory;
+        $this->resourceConnection = $resourceConnection;
+        $this->keywordFactory = $keywordFactory;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $assetIds): array
+    {
+        try {
+            return $this->getAssetKeywords($this->getKeywordsData($assetIds));
+        } catch (\Exception $exception) {
+            $this->logger->critical($exception);
+            throw new IntegrationException(__('Could not retrieve asset keywords.'), $exception);
+        }
+    }
+
+    /**
+     * Load keywords data
+     *
+     * @param array $assetIds
+     * @return array
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function getKeywordsData(array $assetIds): array
```

Comment:
Create DTO (Link between asset id and keywords) for result

Reply:
I don't understand the concern. The public method is returning the `\Magento\MediaGalleryApi\Api\Data\AssetKeywordsInterface[]`